crates/bevy_render/src/renderer/render_resource/render_resource_bindings.rs

```diff
@@ -136,6 +136,7 @@ impl RenderResourceBindings {
         }
     }
 
+    #[allow(clippy::collapsible_match)]
```

Comment:
Is this a bug in the new clippy rule? With the else in line 152, the change clippy suggests would be wrong.

Reply:
The clippy docs state "this lint is not intended to find all cases where nested match patterns can be merged" (https://rust-lang.github.io/rust-clippy/master/index.html#collapsible_match). I interpreted this to mean that in cases where the nested match makes sense sense - like this one where the nested if returns a different value - you can just ignore it.

Reply:
As @DJMcNab just pointed out, the other change for this rule also breaks the logic. The rule seems to suggest this change too easily imo.